handlers/alarms-handler/src/alarmMappings.ts

```diff
@@ -49,22 +54,28 @@ const teamToAppMappings: Record<Team, string[]> = {
 	],
 };
 
-const appToTeamMappings: Record<string, Team> = Object.entries(
-	teamToAppMappings,
-).reduce(
-	(mappings, [team, apps]) => ({
-		...mappings,
-		...apps.reduce((acc, app) => ({ ...acc, [app]: team }), {}),
-	}),
-	{},
-);
+const buildAppToTeamMappings = (): Record<string, Team[]> => {
+	const mappings: Record<string, Team[]> = {};
+
+	for (const [team, apps] of Object.entries(teamToAppMappings)) {
+		for (const app of apps) {
+			const teams = mappings[app] ?? [];
+			teams.push(team as Team);
+
+			mappings[app] = teams;
```

Comment:
I wonder if this is any clearer? Maybe it makes the fact that we're mutating/pushing onto `mappings[app]` a bit more obvious?

```suggestion
			mappings[app] ||= [];
			mappings[app].push(team);
```

Reply:
Ah right yeah good point - final decision is yours!

Reply:
There's also [`??=`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment) which is maybe slightly more correct here.

Reply:
I'm starting to wonder if there are too many operators in the language now!

Reply:
typescript seems to need an intermediary value - when I use either of these operators, it doesn't believe that `mappings[app]` exists -

<img width="337" alt="Screenshot 2024-05-21 at 14 40 39" src="https://github.com/guardian/support-service-lambdas/assets/1513454/b4240ef2-0d0b-4962-b5e1-6b52da9fc5e9">